sdk/eventhub/azure-eventhubs/api_review.py

```diff
@@ -0,0 +1,235 @@
+            partition_id: str = None,
+            owner_level: int = None,
+            prefetch: int = 300,
+            track_last_enqueued_event_properties: bool = False,
+            initial_event_position=None,
+            on_error=None,
+            on_partition_initialize=None,
+            on_partition_close=None
+    ) -> None:
+
+    async def get_properties(self) -> Dict[str, Any]:
+
+    async def get_partition_ids(self) -> List[str]:
+
+    async def get_partition_properties(self, partition: str) -> Dict[str, str]:
+
+    async def close(self) -> None:
+
+
+class PartitionContext(object):
+
+    fully_qualified_namespace  # type: str
+    partition_id  # type: str
+    eventhub_name  # type: str
+    consumer_group_name  # type: str
+    owner_id  # type: str
+
+    def __init__(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        consumer_group_name: str,
+        partition_id: str,
+        owner_id: str,
+        partition_manager: PartitionManager =None) -> None:
+
+    async def update_checkpoint(self, event: EventData):
+
+
+class PartitionManager(ABC):
+
+    async def list_ownership(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        consumer_group_name: str) -> Iterable[Dict[str, Any]]:
+
+    async def claim_ownership(self, ownership_list: Iterable[Dict[str, Any]]) -> Iterable[Dict[str, Any]]:
+
+    async def update_checkpoint(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        consumer_group_name: str,
+        partition_id: str,
+        offset: str,
+        sequence_number: int) -> None:
+
+    async def list_checkpoints(self, fully_qualified_namespace: str, eventhub_name: str, consumer_group_name: str) -> Iterable[Dict[str,Any]]:
+
+
+class EventData(object):
+
+    message  # type: uamqp.Message
+
+    def __init__(self, body: Union[str, bytes] =None) -> None:
+
+    def __str__(self) -> str:
+
+    @property
+    def sequence_number(self) -> int:
+
+    @property
+    def offset(self) -> str:
+
+    @property
+    def enqueued_time(self) -> datetime.datetime:
+    
+    @property
+    def partition_key(self) -> bytes:
+
+    @property
+    def application_properties(self) -> dict[str, Any]:
+
+    @property
+    def system_properties(self) -> dict[str, Any]:
+
+    @property
+    def body(self) -> Union[bytes, Iterable[bytes]]:
+
+    @property
+    def last_enqueued_event_properties(self) -> dict[str, Any]:
+
+    def body_as_str(self, encoding='UTF-8') -> str:
+
+    def body_as_json(self, encoding='UTF-8') -> dict[str, Any]:
+
+    def encode_message(self) -> bytes:
+
+
+class EventDataBatch(object):
+
+    message  # type: uamqp.BatchMessage
+    max_size  # type: int
+
+    def __init__(self, max_size: int=None, partition_key: str=None) -> None:
+    
+    def __len__(self) -> int:
+
+    @property
+    def size(self) -> int:
+
+    def try_add(self, event_data: EventData) -> None:
```

Comment:
I understand there's some history (thanks Yijun and Adam for OOB context) with respect to try_add and returning Boolean vs exceptioning.  While it's more pythonic to throw, I might suggest calling it just "add" in that case, since there's not really a try aspect.

The downside to that is if we break with other libs.  Heretical suggestion that there's always the option to support both signatures, but I feel like I'm probably walking into a beartrap by saying that, so take this as "up for discussion if others don't have strong opinions"

Reply:
We could also renamed to `append` to create a more list-like API.

Reply:
To discuss internally.